packages/flutter_tools/bin/xcode_debug.js

```diff
@@ -0,0 +1,519 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/**
+ * @fileoverview OSA Script to interact with Xcode. Functionality includes
+ * checking if a given project is open in Xcode, starting a debug session for
+ * a given project, and stopping a debug session for a given project.
+ */
+
+'use strict';
+
+function run(args_array = []) {
+  let args;
+  try {
+    args = new CommandArguments(args_array);
+  } catch (e) {
+    return new RunJsonResponse(false, `Failed to parse arguments: ${e}`);
+  }
+
+  const xcodeResult = getXcode(args);
+  if (xcodeResult.error != null) {
+    return new RunJsonResponse(false, xcodeResult.error).stringify();
+  }
+  const xcode = xcodeResult.result;
+
+  if (args.command === 'project-opened') {
+    const result = getWorkspace(xcode, args);
+    return new RunJsonResponse(result.error == null, result.error).stringify();
+  } else if (args.command === 'debug') {
+    const result = debugApp(xcode, args);
+    return new RunJsonResponse(result.error == null, result.error, result.result).stringify();
+  } else if (args.command === 'stop') {
+    const result = stopApp(xcode, args);
+    return new RunJsonResponse(result.error == null, result.error).stringify();
+  } else {
+    return new RunJsonResponse(false, 'Unknown command').stringify();
+  }
+}
+
+/**
+ * Parsed and validated arguments passed from the command line.
+ */
+class CommandArguments {
+  /**
+   *
+   * @param {!Array<string>} args List of arguments passed from the command line.
```

Comment:
what does the exclamation point here mean?

Reply:
It means it's expected to be non-null (https://google.github.io/styleguide/jsguide.html#jsdoc-nullability)

Reply:
:+1: